File: src/WalletApp.java
import model.History;
import model.Note;
import model.OperationType;
import model.Wallet;
import service.WalletService;
import view.BarView;
import view.HistoryView;
import view.MenuView;

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class WalletApp {
    public static void run () {
        Scanner sc = new Scanner(System.in);
        Wallet wallet = new Wallet();
        List<History> historyList = new ArrayList<>();
        List<Note> noteList = new ArrayList<>();

        System.out.println("Welcome");

        boolean conditionRunning = true;
        while (conditionRunning) {
            BarView.bar("Menu");
            MenuView.printMenu("Status", "Deposit", "Withdraw", "History", "Note", "Exit");
            while (!sc.hasNextInt()) {
                System.out.println("Please enter a valid number");
                sc.next();
            }
            int choice = sc.nextInt();
            switch (choice) {
                case 1:
                    WalletService.printWallet(wallet);
                    historyList.add(new History(OperationType.STATUS, new Wallet(wallet.getSold())));
                    break;
                case 2:
                    WalletService.deposit(wallet, sc);
                    historyList.add(new History(OperationType.DEPOSIT, new Wallet(wallet.getSold())));
                    break;
                case 3:
                    WalletService.withdrawal(wallet, sc);
                    historyList.add(new History(OperationType.WITHDRAW, new Wallet(wallet.getSold())));
                    break;
                case 4:
                    HistoryView.printHistory(historyList);
                    break;
                case 5:
                    NoteApp.run(noteList, sc);
                    break;
                case 6:
                    conditionRunning = false;
                    System.out.println("Bye");
                    break;
                default:
                    System.out.println("RETRY");
            }
        }
    }
}

File: src/wallet.app.ts
import { OperationHistory } from "./model/operation.history.model";
import { OperationType } from "./model/operation.type";
import { Wallet } from "./model/wallet.model";
import { deposit, printWallet, withdrawal } from "./service/wallet.service";
import { closeMyPrompt, myPrompt } from "./util/my.prompt.util";
import { bar } from "./view/bar.view";
import { printHistory } from "./view/history.view";
import { printMenu } from "./view/menu.view";

export const run = async () => {
  let wallet: Wallet = new Wallet(0);
  let historyList: OperationHistory[] = [];

  console.log("Welcome");

  let conditionRunning: boolean = true;
  while (conditionRunning) {
    printMenu("Status", "Deposit", "Withdraw", "History", "Exit");
    let input = await myPrompt("Choice: ");
    let choice: number = Number(input);
    while (!choice) {
      input = await myPrompt("Please enter a valid number");
    }
    switch (choice) {
      case 1:
        bar("Status")
        printWallet(wallet);
        historyList.push(
          new OperationHistory(
            OperationType.STATUS,
            new Wallet(wallet.getSold())
          )
        );
        break;
      case 2:
        bar("Deposit")
        await deposit(wallet)
          .then((res) => {
            console.log(res);
            historyList.push(
              new OperationHistory(
                OperationType.DEPOSIT,
                new Wallet(wallet.getSold())
              )
            );
          })
          .catch((err) => console.log(err.message));
        break;
      case 3:
        bar("Withdraw");
        await withdrawal(wallet)
          .then((res) => {
            console.log(res);
            historyList.push(
              new OperationHistory(
                OperationType.WITHDRAW,
                new Wallet(wallet.getSold())
              )
            );
          })
          .catch((err) => console.log(err.message));
        break;
      case 4:
        printHistory(historyList);
        break;
      case 5:
        conditionRunning = false;
        closeMyPrompt();
        console.log("Bye");
        break;
      default:
        console.log("RETRY");
    }
  }
};
